In [1]:
import gurobipy as gp

Neste exemplo temos 3 produtos onde cada um com seu lucro unitários, sendo:

 Produto  | Lucro Unitário  | Variável de Decisão  | Valores |
:---:|:---:|:---:|---:
  A | 12  | x1  | ?  
  B | 18  | x2  | ? 
  C | 22  | x3  | ? 


Na tabela abaixo temos o total de produção por maquina, é possivel observar que o produto **A** consume 1,5 da máquina 01, nenhum tempo da máquina 02 e 1,2 da máquina 03.

 Produto  | Máquina 01  | Máquina 02  | Máquina 03 
:---:|:---:|:---:|:---:
  A | 1.5  | 0  | 1.2  
  B | 0  | 2.2  | 2 
  C | 1.2  | 1.4  | 2.4 
  **Tempo Disponível** | **120**  | **200**  | **250** 
  **Tempo Usado** | **0**  | **0**  | **0** 

In [20]:
# Criação do modelo
m = gp.Model("MIX de Produção")

# Variavel de decisão
x1 = m.addVar()
x2 = m.addVar()
x3 = m.addVar()

# Função Objetivo do problema
# sense é o sentido do modelo
m.setObjective(12 * x1  + 18 * x2 + 22 * x3, sense=gp.GRB.MAXIMIZE)

In [21]:
# configurações de restrições
c1 = m.addConstr(1.5 * x1 + 1.2 * x3 <= 120)
c2 = m.addConstr(2.2 * x2 + 1.4 * x3 <= 200)
c3 = m.addConstr(1.2 * x1 + 2 * x2 + 2.4 * x3 <= 250)

In [22]:
# Resolução do modelo
m.optimize()

Gurobi Optimizer version 9.1.0 build v9.1.0rc0 (mac64)
Thread count: 4 physical cores, 8 logical processors, using up to 8 threads
Optimize a model with 3 rows, 3 columns and 7 nonzeros
Model fingerprint: 0xa81d5c28
Coefficient statistics:
  Matrix range     [1e+00, 2e+00]
  Objective range  [1e+01, 2e+01]
  Bounds range     [0e+00, 0e+00]
  RHS range        [1e+02, 2e+02]
Presolve time: 0.01s
Presolved: 3 rows, 3 columns, 7 nonzeros

Iteration    Objective       Primal Inf.    Dual Inf.      Time
       0    5.2000000e+31   5.000000e+30   5.200000e+01      0s
       5    2.3460000e+03   0.000000e+00   0.000000e+00      0s

Solved in 5 iterations and 0.01 seconds
Optimal objective  2.346000000e+03


In [17]:
print("Produto A: {0}".format(x1.X))
print("Produto B: {0}".format(x2.X))
print("Produto C: {0}".format(x3.X))

Produto A: 80.0
Produto B: 77.0
Produto C: 0.0


| Valores  |
| :---:  |
|  80  |
|  77  |
|  0  |

In [19]:
time_m1 = 120 - c1.Slack
time_m2 = 200 - c2.Slack
time_m3 = 250 - c3.Slack

print("Máquina 1: {0}".format(time_m1))
print("Máquina 2: {0}".format(time_m2))
print("Máquina 3: {0}".format(time_m3))

Máquina 1: 120.0
Máquina 2: 169.4
Máquina 3: 250.0


 Produto  | Máquina 01  | Máquina 02  | Máquina 03 
:---|:---:|:---:|:---:
**Tempo Disponível** | **120**  | **200**  | **250** 
**Tempo Usado** | **120**  | **169.4**  | **250** 
**Sobra** | **0**  | **30.6**  | **0** 